nvflare/private/fed/client/training_cmds.py

```diff
@@ -117,19 +118,21 @@ def process(self, req: Message, app_ctx) -> Message:
         if not job_meta:
             return error_reply("missing job meta")
 
-        err = engine.deploy_app(
-            app_name=app_name, job_id=job_id, job_meta=job_meta, client_name=client_name, app_data=req.body
-        )
-        if err:
-            return error_reply(err)
-
         kv_list = parse_vars(engine.args.set)
         secure_train = kv_list.get("secure_train", True)
-        if secure_train:
+        from_hub_site = job_meta.get(JobMetaKey.FROM_HUB_SITE.value)
+        if secure_train and not from_hub_site:
```

Comment:
suggest `from_hub_site = job_meta.get(JobMetaKey.FROM_HUB_SITE.value, False)`

Need to make sure "job_meta[JobMetaKey.FROM_HUB_SITE]" returns a boolean type not string type.
If returns a string "False" then it will be evaluate to True.